### PR TITLE
Fix: Add support for escape char in JSX.

### PR DESCRIPTION
**Pull request by sstern6**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Additional spaces were being accounted for in the logic, I implemented a fix, trimming the escape character to have an accurate representation of the escapes given the current logic.

**Is there anything you'd like reviewers to focus on?**
My implementation, is there a better way to remove whitespace that may be more efficient?
Also, in the JSDOC for the function `function validate(escapes, node, match) {`, the JSDocs state, `* @param {string} match - string slice to validate.`.  From what I have seen isnt match an array, not a string?

This PR fixes: https://github.com/eslint/eslint/issues/7460

Thank you

### CODE REVIEW

**Comment by mysticatea** [comment]
Thank you for the contributing!

However, I don't think this fix is the right direction.
This rule should report the escape of a space since it's unnecessary escape. Could you add the test for it?

- `"\ "`
- `/\ /`

To fix that issue, I think we should modify [this check](https://github.com/eslint/eslint/blob/ee3bcea71fe8f826e7c0494f5388ad80542fbc35/lib/rules/no-useless-escape.js#L143-L149) to ignore JSX.

```js
                /*
                 * JSXAttribute doesn't have any escape sequence: https://facebook.github.io/jsx/.
                 * In addition, backticks are not supported by JSX yet: https://github.com/facebook/jsx/issues/25.
                 */
                if (node.parent.type === "JSXAttribute" || node.parent.type === "JSXElement") {
                    return;
                }
```

**Comment by sstern6** [comment]
@mysticatea thank you for your feedback, will have this back up later today/tomorrow.

File: tests/lib/rules/no-useless-escape.js [hunk 1]
```diff
@@ -47,6 +47,8 @@ ruleTester.run("no-useless-escape", rule, {
         "var foo = '\\\n';",
         "var foo = '\\\r\n';",
         {code: "<foo attr=\"\\d\"/>", parserOptions: {ecmaFeatures: {jsx: true}}},
+        {code: "<div> Testing: \ </div>", parserOptions: {ecmaFeatures: {jsx: true}}},
```
Comment (sstern6) [inline tests/lib/rules/no-useless-escape.js]: Thanks, apologize, I was getting confused with the escapes in the test file!

**Comment by sstern6** [comment]
@mysticatea re-pushed with updated changes.  Implemented the logic you recommended.  Please let me know if theres anything else I have missed or need to add.

Thanks

**Comment by sstern6** [comment]
@nzakas saw you removed the `CLA: valid` tag, but the PR isnt prompting me to resign the CLA.

Is there anything else you need from me on my end?

Thanks

**Comment by kaicataldo** [comment]
@sstern6 The check is now the license/cla check that gets run with the CI tests - looks good to me!

**Pull request merged**